src/utils/mardyn_assert.h

```diff
@@ -10,20 +10,25 @@
 
 #include "Logger.h"
 
-inline void mardyn_exit(int code) {
+// Macro to wrap mardyn_exit and pass the caller function, file and line
+#define MARDYN_EXIT(code) mardyn_exit(code, __FUNCTION__, __FILE__, __LINE__)
+
+inline void mardyn_exit(int code, const char* caller_function, const char* file, int line) {
+	std::cerr << "Exit with exit code " << code
+			  << " called from function \"" << caller_function
+			  << "\" in file " << file << ":" << line << std::endl;
```

Comment:
```suggestion
	std::cerr << "Exit with exit code " << code
			  << "\ncalled from function `" << caller_function
			  << "`\nin file " << file << ":" << line << std::endl;
```
Just some formatting suggestions.

Reply:
Also: Why not use the logger here?

Reply:
For multiple line error messages, I prefer printing them with the same indentation. If the logger is used, it adds the time etc, so it will look like:
```log
ERROR:  20240101T010101 0.001065 [0]    ERROR parsing all chars of "10a0000" from tag "<steps>" in xml file
This might be the result of using a float while an integer is expected.
```

Reply:
The downsides of not using the logger are:
- Exit messages might be out of sync with other messages. This can more easily lead to overlapping messages (=data races) or even log messages appearing after the exit message even though they were issued before the error.
- By default the logger prints to `std::out`. If the user only redirects this to an output file they might loose the error output because it unexpectedly is printed to a different system stream.
- The error has no timestamp.

If you dislike the indent thing feel free to just put everything on one line again, but then maybe insert a `:` before each information so that they are easier splittable.

```suggestion
	std::cerr << "Exit with exit code: " << code
				  << " called from function: " << caller_function
				  << " in file: " << file << ":" << line << std::endl;
```

Reply:
I solved it now that way: https://github.com/ls1mardyn/ls1-mardyn/pull/341/commits/c80fde886adb60c1dc9a3387d2c4fc1ebb8317f0